homeworks/oleg.dzhygil_dzhygil/homework_5/homework_5.js

```diff
@@ -0,0 +1,81 @@
+// Task 1
+
+const arrayFirst = [10, -10, 10, -10, 10];
+const arraySecond = [0, 0, 0, 0, 0];
+const arrayThird = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const sumOfArray = (array) => {
+    let number = 0;
+    return array.map((value) => {
+        number += value;
+        return number;
+    });
+};
+
+console.log(sumOfArray(arrayFirst));
+console.log(sumOfArray(arraySecond));
+console.log(sumOfArray(arrayThird));
+
+
+// Task 2
+
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+
+
+const sortKey = (arr) => {
+    const obj = {};
+    arr.forEach((x) => {
+        if (x in obj) {
+            obj[x] += 1;
+        } else {
+            obj[x] = 1;
+        }
+    });
+    return obj;
+};
+console.log(sortKey(classes));
+
+// Task 3
+
+const obj = { b: 'c', d: { e: 'f' } };
+
+function innerCopy() {
+    const newObject = {};
+    Object.keys(obj).forEach((key) => {
+        newObject[key] = obj[key];
+    });
```

Comment:
this is not deep copy once again

if you modify `d.e` property of the initial object in will mutate copy as well and vise versa

Reply:
done